source/fundamentals/crud/write-operations/bulk.txt

```diff
@@ -366,6 +370,346 @@ the bulk operation:
+
+   * - ``SetHint()`` 
+     - | The index to use to scan for documents.
+
+   * - ``SetReplacement()`` 
+     - | The document to replace the matched document with.
+
+   * - ``SetSort()`` 
+     - | The sort order for matching documents. The replace operation
+         replaces only the first document according to the sort criteria.
+
+   * - ``SetUpsert()`` 
+     - | Whether to insert a new document if the :ref:`query filter <golang-query-filter-definition>` doesn't match any documents.
+
+This example creates ``ClientReplaceOneModel`` instances to define
+the following operations:
+
+- Replace operation on the ``books`` collection to replace a 
+  document in which the ``title`` value is ``"Lucy"``
+
+- Replace operation on the ``poems`` collection to replace a 
+  document in which the ``title`` value is ``"Song of Myself"``
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/bulkOps.go
+   :language: go
+   :dedent:
+   :start-after: begin bulk replace model client
+   :end-before: end bulk replace model client
+
+ClientUpdateOneModel
+````````````````````
+
+To define an update operation for a bulk write, create
+a ``ClientUpdateOneModel`` specifying the document you
+want to update and an :ref:`update document <golang-update-document>`.
+To update multiple documents, use a ``ClientUpdateManyModel``.
+
+You can specify the behavior of each of the update models
+by using the following methods:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``SetArrayFilters()`` 
+     - | The array elements the update applies to.
+
+   * - ``SetCollation()`` 
+     - | The type of language collation to use when sorting results.
+
+   * - ``SetFilter()`` 
+     - | The :ref:`query filter <golang-query-filter-definition>` specifying which document to update.
+
+   * - ``SetHint()`` 
+     - | The index to use to scan for documents.
+
+   * - ``SetSort()`` 
+     - | The criteria to use when ordering matching documents.
+         This method is not available for the ``ClientUpdateManyModel`` class.
+
+   * - ``SetUpdate()`` 
+     - | The modifications to apply on the matched documents.
+
+   * - ``SetUpsert()`` 
+     - | Whether to insert a new document if the :ref:`query filter <golang-query-filter-definition>`
+         doesn't match any documents.
+
+This example creates ``ClientUpdateOneModel`` instances to define
+the following operations:
+
+- Update operation on the ``books`` collection to update a 
+  document in which the ``author`` value is ``"Elena Ferrante"``
+
+- Update operation on the ``poems`` collection to update a 
+  document in which the ``author`` value is ``"Ada Limon"``
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/bulkOps.go
+   :language: go
+   :dedent:
+   :start-after: begin bulk update model client
+   :end-before: end bulk update model client
+
```

Comment:
S: add a note that describes updatemany model as an option to update all matching documents